src/opera/threading/node_executor.py

```diff
@@ -60,7 +56,7 @@ def process_results(self, results):
             try:
                 future.result()
                 self.processed_nodes.remove(node_id)
-            except Exception as e:
+            except (CancelledError, TimeoutError) as e:
```

Comment:
Hi @sstanovnik. I was checking the code and stumbled upon this change in exception handling. What was the idea behind it?

Reply:
I wanted to avoid having `except Exception` everywhere, so I checked what errors can predictably be thrown. Is this not catching something? If so, this is perfect, since it exposed a code path I didn't know about :)

Reply:
I agree, it's great to have exact types of errors to see what exceptions we can expect in certain part of code, so @alexmaslenn if you found additional exception you can add it to the list.

Overall `opera`'s source code lacks typing in a lot of places and we'll try to gradually add types everywhere to be more developer-friendly.

Reply:
@sstanovnik yes in `future.result()` we finalize execution of asynchronous operation and technically it can raise any exception that underlying function raises. I think is is fixed now.

Reply:
@alexmaslenn ok if this is the case we can then revert back.

Reply:
@anzoman I had already fixed this in the last PR